tests/syft/core/pointer/garbage_collection/gc_strategies_test.py

```diff
@@ -0,0 +1,91 @@
+# third party
+import torch
+
+# syft absolute
+import syft as sy
+from syft.core.pointer.garbage_collection import GCBatched
+from syft.core.pointer.garbage_collection import GCSimple
+from syft.core.pointer.garbage_collection import GarbageCollection
+from syft.core.pointer.garbage_collection import gc_get_default_strategy
+from syft.core.pointer.garbage_collection import gc_set_default_strategy
+
+
+def test_gc_simple_strategy(node: sy.VirtualMachine) -> None:
+    client = node.get_client()
+
+    x = torch.tensor([1, 2, 3, 4])
+    ptr = x.send(client, pointable=False)
+
+    assert len(node.store) == 1
+
+    del ptr
+
+    assert len(node.store) == 0
+
+
+def test_gc_batched_strategy_setter(node: sy.VirtualMachine) -> None:
+
+    client = node.get_client()
+    client.gc.gc_strategy = GCBatched(threshold=10)
+
+    x = torch.tensor([1, 2, 3, 4])
+
+    for _ in range(9):
+        x.send(client, pointable=False)
+
+    assert len(node.store) == 9
+
+    x.send(client, pointable=False)
+
+    assert len(node.store) == 0
+
+
+def test_gc_batched_strategy_gc_constructor(node: sy.VirtualMachine) -> None:
+    client = node.get_client()
+    client.gc = GarbageCollection("gcbatched", 5)
```

Comment:
@tudorcebere this is the way that I would think most people would use the gc strategy

Reply:
I like this as well, but we need really nice docs on it on how to actually use them/how to implement one. Most likely, people won't be aware of this feature, but it's nice to have the option to do it. (Maybe we want to cut network usage and never free anything, stuff like that).

Reply:
Yep - I would do that in the ```__init__.py``` file from ```garbage_collection``` dir